### PR TITLE
Add SQLite backend

### PR DESCRIPTION
**Pull request by wildwestrom**
I just have to update the readme and it's probably good.

### CODE REVIEW

File: test/duratom/core_test.clj [hunk 1]
```diff
@@ -283,6 +283,70 @@
...
+    (when async?
+      (common* (duragent :sqlite-db
+                         :db-config db-spec
+                         :table-name table-name
+                         :row-id 0
+                         :init init)
+               #(some? (ut/get-sql-value db-spec table-name 0 ut/read-edn-string))
+               async?))
+    ;; manually delete sqlite file to clean up
+    (.delete (io/file subname))))
```
Comment (jimpil) [inline test/duratom/core_test.clj]: prefer `io/delete-file` similar to line 290

File: src/clojure/duratom/utils.clj [hunk 2]
```diff
@@ -202,16 +202,21 @@
                                                         [[:id :int] [:value col-type]]))
     (catch BatchUpdateException _ '(0)))) ;; table already exists!
 
-(defn get-pgsql-value [db table-name row-id read-it!]
```
Comment (jimpil) [inline src/clojure/duratom/utils.clj]: revert this renaming, and simply add the following:
```clj
(def get-sqlite-value get-pgsql-value)
```

**Review by jimpil — CHANGES_REQUESTED** [review]

File: src/clojure/duratom/core.clj [hunk 3]
```diff
@@ -485,6 +517,12 @@
              rw default-postgres-rw}}]
   (postgres-atom db-config table-name row-id lock init rw))
 
+(defmethod duratom :sqlite-db
+  [_ & {:keys [db-config table-name row-id init lock rw]
+        :or {lock (ReentrantLock.)
+             rw default-postgres-rw}}]
```
Comment (jimpil) [inline src/clojure/duratom/core.clj]: use `default-sqlite-rw`

File: src/clojure/duratom/core.clj [hunk 4]
```diff
@@ -584,7 +622,22 @@
...
+                                    table-name))
+                              row-id
+                              (:read rw)
+                              (:write rw))]
+    (duragent* init meta (:error-handler rw) make-backend)))
+
+(defmethod duragent :sqlite-db
+  [_ & {:keys [db-config table-name row-id init lock rw meta]
+        :or {lock (ReentrantLock.)
+             rw default-postgres-rw}}]
```
Comment (jimpil) [inline src/clojure/duratom/core.clj]: and here

**Comment by jimpil** [comment]
Hi @wildwestrom, everything ok here? it's been a week since you opened this PR, and I've not heard back since...do you require any help addressing my comments?

**Comment by wildwestrom** [comment]
My bad, just haven't gotten to it yet.

**Comment by wildwestrom** [comment]
Ok, that should be good.

File: src/clojure/duratom/core.clj [hunk 5]
```diff
@@ -581,10 +619,25 @@
 (defmethod duragent :postgres-db
   [_ & {:keys [db-config table-name row-id init lock rw meta]
         :or {lock (ReentrantLock.)
-             rw default-postgres-rw}}]
+             rw default-sqlite-rw}}]
```
Comment (jimpil) [inline src/clojure/duratom/core.clj]: i think should be reverted  - this is `postgres-db` duragent

**Review by jimpil — CHANGES_REQUESTED** [review]
Ok, looks good - just one little thing (which  seems to have happened by accident).

**Review by jimpil — APPROVED** [review]
Great stuff! Thanks for this...You can expect a new release by the end of the weekend 👍

**Comment by jimpil** [comment]
FYI, `duratom 0.5.4` is up on clojars 🚀

**Pull request merged**